camera_simulator/camera_simulator.py

```diff
@@ -53,36 +54,36 @@ def __init__(self, **kwargs):
         if self.type == "video":
             try:
                 self.vc = cv2.VideoCapture(kwargs["path"])
+                self.vc.set(cv2.CAP_PROP_POS_MSEC, kwargs["start"])
             except:
                 print("End of file")
 
-            timer_period = 0.05  # seconds
-            self.timer = self.create_timer(timer_period, self.image_callback)
+            video_fps = self.vc.get(cv2.CAP_PROP_FPS)
+            self.get_logger().info(f"Publishing image with {video_fps} fps")
+
+            self.timer = self.create_timer(1.0 / video_fps, self.image_callback)
         else:
             for image_path in natsorted(os.listdir(kwargs["path"]), key=lambda y: y.lower()):
                 if image_path.endswith(".jpg") or image_path.endswith(".jpeg") or image_path.endswith(".png"):
                     self.image_callback(os.path.join(kwargs["path"], image_path))
-            rclpy.get_logger().info("All images have been published")
-
+            self.get_logger().info("All images have been published")
 
     def image_callback(self, image_path=None):
-        self.get_logger().info('Publishing image from simulator')
-
         if self.type == "video":
             rval, image = self.vc.read()
         elif image_path:
             image = cv2.imread(image_path)
         else:
-            rclpy.get_logger().error("Image path is none.")
+            self.get_logger().error("Image path is none.")
             raise ValueError()
 
         time_msg = self.get_time_msg()
-        img_msg = self.get_image_msg(image, time_msg) # Convert the image to a message
+        img_msg = self.get_image_msg(image, time_msg)  # Convert the image to a message
 
-        camera_info_msg = self.get_camera_info(time_msg)
+        # camera_info_msg = self.get_camera_info(time_msg)
 
         self.image_publisher_.publish(img_msg)
-        self.camera_info_publisher_.publish(camera_info_msg)
+        # self.camera_info_publisher_.publish(camera_info_msg)
```

Comment:
I'll merge, because I saw that I try to read the calibration file for each message, which is kinda insane. After that I'll re-enable the camera calibration message, because that is quite important to have :)

Reply:
Ah sorry, forgot about that. Was a bit messy indeed :D